pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go

```diff
@@ -226,9 +252,13 @@ func (r *ReconcileAmazonCloudIntegration) Reconcile(ctx context.Context, request
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileAmazonCloudIntegration) SetDegraded(reason string, err error, log logr.Logger) {
-	log.Error(err, reason)
-	r.status.SetDegraded(reason, err.Error())
+func (r *ReconcileAmazonCloudIntegration) SetDegraded(reason operatorv1.TigeraStatusReason, message string, err error, log logr.Logger) {
```

Comment:
Rather than having a duplicate and identical method for each reconciler, maybe you can have one func in status.go that does:
```
func SetDegraded(stat status.StatusManager, reason operatorv1.TigeraStatusReason, message string, err error, log logr.Logger) {
	log.WithValues(string(reason), message).Error(err, string(reason))
	errormsg := ""
	if err != nil {
		errormsg = err.Error()
	}
	stat.SetDegraded(string(reason), fmt.Sprintf("%s - Error: %s", message, errormsg))
}
```

Reply:
Done